Source/WebCore/rendering/style/RenderStyle.cpp

```diff
@@ -1281,19 +1281,19 @@ bool RenderStyle::changeRequiresRepaint(const RenderStyle& other, OptionSet<Styl
         return true;
 
 
-    if (m_nonInheritedData.ptr() != other.m_nonInheritedData.ptr()) {
-        if (m_nonInheritedData->backgroundData.ptr() != other.m_nonInheritedData->backgroundData.ptr()) {
-            if (!m_nonInheritedData->backgroundData->isEquivalentForPainting(*other.m_nonInheritedData->backgroundData, currentColorDiffers))
-                return true;
-        }
+    bool nonInheritedDataDiffers = m_nonInheritedData.ptr() != other.m_nonInheritedData.ptr();
```

Comment:
`nonInheritedDataDiffers` is not an accurate name; even if the pointers are different, the data might be the same. We have `arePointingToEqualData()` to help with this comparison.